.openpublishing.redirection.json

```diff
@@ -2780,9 +2780,234 @@
         },
         {
             "source_path": "docs/framework/xaml-services/{}-escape-sequence-markup-extension.md",
-            "redirect_url": "/dotnet/framework/xaml-services/escape-sequence-markup-extension",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/escape-sequence-markup-extension",
+            "redirect_document_id": false
+        },
+        {
+            "source_path": "docs/framework/xaml-services/built-in-types-for-common-xaml-language-primitives.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/built-in-types-for-common-xaml-language-primitives",
+            "redirect_document_id": true
+        },
+        {
+            "source_path": "docs/framework/xaml-services/collections-and-collection-types-for-xaml.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/collections-and-collection-types-for-xaml",
+            "redirect_document_id": true
+        },
+        {
+            "source_path": "docs/framework/xaml-services/default-xaml-schema-context-and-wpf-xaml-schema-context.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/default-xaml-schema-context-and-wpf-xaml-schema-context",
+            "redirect_document_id": true
+        },
+        {
+            "source_path": "docs/framework/xaml-services/defining-custom-types-for-use-with-net-framework-xaml-services.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/defining-custom-types-for-use-with-net-framework-xaml-services",
+            "redirect_document_id": true
+        },
+        {
+            "source_path": "docs/framework/xaml-services/escape-sequence-markup-extension.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/escape-sequence-markup-extension",
+            "redirect_document_id": true
+        },
+        {
+            "source_path": "docs/framework/xaml-services/generics-in-xaml.md",
+            "redirect_url": "/dotnet/desktop-wpf/xaml-services/generics-in-xaml",
```

Comment:
now since you're doing a file-based redirection you could also improve the file names to use better naming conventions like not having small words like in or repeating words in path.